lib/routes/seagullApi.js

```diff
@@ -288,32 +289,42 @@ module.exports = function (crudHandler, userApiClient, gatekeeperClient, metrics
           try {
             const userIds = _.keys(mergedUserPermissions);
             // Break requests for users into chunks of 200, so that the query parameter doesn't get too long
-            const results = await mapLimit(_.chunk(userIds, 200), 5, async usersChunk => {
-              const getUsers = util.promisify(userApiClient.getUsersWithIds);
-              try {
-                const users = await getUsers(usersChunk);
-                if (!users) {
-                  throw new Error(`No users returned for user id ${targetUserId}`);
-                } else {
-                  return users;
-                }
-              } catch (error) {
-                throw new Error(`Error getting users: ${error}`);
+            const getUsers = util.promisify(userApiClient.getUsersWithIds);
+            const results = await mapLimit(_.chunk(userIds, 200), 5, async (usersChunk) => {
+              const users = await getUsers(usersChunk);
+              if (!users) {
+                throw new Error(`No users returned for user id ${targetUserId}`);
+              } else {
+                return users;
               }
             });
             userProfiles = _.flatten(results);
 
             if (userProfiles.length !== userIds.length) {
-              throw new Error('Received different number of results than we queried for');
+              const missingProfiles = [];
+              if (userProfiles.length < userIds.length) {
+                for (const userId of userIds) {
+                  let found = false;
+                  for (const profile of userProfiles) {
+                    if (profile.userid === userId) {
+                      found = true;
```

Comment:
we can break the loop once found
```suggestion
                      found = true;
                      break;
```

Reply:
I don't understand why I didn't see this...
Good catch